src/deployment/mod.rs

```diff
@@ -0,0 +1,1086 @@
+            contract_id.clone(),
+            contract.source.clone(),
+            parser_version,
+        );
+        contract_asts.insert(contract_id.clone(), ast);
+        contract_diags.insert(contract_id.clone(), diags);
+    }
+
+    let dependencies =
+        ASTDependencyDetector::detect_dependencies(&contract_asts, &requirements_asts);
+
+    let mut dependencies = match dependencies {
+        Ok(dependencies) => dependencies,
+        Err(_) => {
+            return Err(format!("unable to detect dependencies"));
+        }
+    };
+
+    for contract_id in boot_contracts_ids.into_iter() {
+        dependencies.insert(contract_id.clone(), DependencySet::new());
+    }
+
+    dependencies.extend(requirements_deps);
+
+    let ordered_contracts_ids = match ASTDependencyDetector::order_contracts(&dependencies) {
+        Ok(ordered_contracts_ids) => ordered_contracts_ids
+            .into_iter()
+            .map(|c| c.clone())
+            .collect::<Vec<_>>(),
+        Err(e) => return Err(format!("unable to order contracts {}", e)),
+    };
+
+    for contract_id in ordered_contracts_ids.into_iter() {
+        if requirements_asts.contains_key(&contract_id) {
+            continue;
+        }
+        let data = contracts
+            .remove(&contract_id)
+            .expect("unable to retrieve contract");
+        contracts_map.insert(
+            contract_id.clone(),
+            (data.source.clone(), data.relative_path.clone()),
+        );
+        let tx = if network.is_none() {
+            TransactionSpecification::EmulatedContractPublish(data)
+        } else {
+            TransactionSpecification::ContractPublish(ContractPublishSpecification {
+                contract_name: data.contract_name.clone(),
+                expected_sender: data.emulated_sender.clone(),
+                relative_path: data.relative_path.clone(),
+                source: data.source.clone(),
+                cost: deployment_fee_rate
+                    .saturating_mul(data.source.as_bytes().len().try_into().unwrap()),
+            })
+        };
+
+        transactions.push(tx);
+    }
+
+    let _tx_chain_limit = 25;
+
+    let mut batches = vec![];
+    for (id, transactions) in transactions.chunks(25).enumerate() {
+        batches.push(TransactionsBatchSpecification {
+            id: id,
+            transactions: transactions.to_vec(),
+        })
+    }
+
+    let mut wallets = vec![];
+    for (name, account) in chain_config.accounts.into_iter() {
+        let address = match PrincipalData::parse_standard_principal(&account.address) {
+            Ok(res) => res,
+            Err(_) => {
+                return Err(format!(
+                    "unable to parse wallet {} in a valid Stacks address",
+                    account.address
+                ))
+            }
+        };
+
+        wallets.push(WalletSpecification {
+            name,
+            address,
+            balance: account.balance.into(),
+        });
+    }
```

Comment:
Should this be gated by `if network.is_none()`? It doesn't really matter, since it won't get used otherwise, but might make it more clear.

Reply:
Addressed with latest